spec/services/atmosphere/billing_service_spec.rb

```diff
@@ -255,15 +330,20 @@
       # Standard non-shared appliance
       appl1 = create(:appliance, fund: cs_fund, appliance_set: wf, appliance_type: not_shareable_appl_type, appliance_configuration_instance: config_inst1, virtual_machines: [not_shareable_vm1])
 
-      cost_unit = appl1.virtual_machines.first.virtual_machine_flavor.hourly_cost # Cost per 1h of use for a single (full) VM
+      vmf = appl1.virtual_machines.first.virtual_machine_flavor
+      vmf.set_hourly_cost_for(Atmosphere::OSFamily.first, 10)
+      vmf.reload
```

Comment:
OK, I think this can be resolved by adding the obscure `inverse_of:` flag to the ActiveRecord relation. This establishes an identity-based link between A and B so that when A "has many" Bs and B is updated, the `A.Bs` reference will point to the updated object instead of the in-memory (outdated) version of A.Bs.

Added experimentally to `VirtualMachineFlavor` and 'VirtualMachineFlavorOSFamily` but I think this could help in many other places too.